src/plugins/intel_npu/src/plugin/npuw/partitioning/patterns/opt.cpp

```diff
@@ -1292,6 +1281,42 @@ CompressDictMatMulf32::CompressDictMatMulf32(Context::Ref ctx) {
     register_matcher(std::make_shared<opp::Matcher>(res, "OptCompressDictMatMulf32"), std::move(callback));
 }
 
+//
+// FROM:
+//     Param [1x4x1023x64] -> ShapeOf -> ...
+//
+// TO:
+//     Const [4] -> ...
+//
+ShapeOfToConst::ShapeOfToConst(const std::shared_ptr<ov::Model>& model) {
+    auto key_val = opp::wrap_type<ov::op::v0::Parameter>();
+    auto shapeof = opp::wrap_type<ov::op::v3::ShapeOf>({key_val});
+
+    // Note: Use [=] to make sure the above objects stay alive in the callback
+    auto callback = [=](ov::pass::pattern::Matcher& m) {
+        auto& node_to_output = m.get_pattern_value_map();
+
+        auto matched_node_key_val = node_to_output.at(key_val).get_node_shared_ptr();
+        auto matched_node_shapeof = node_to_output.at(shapeof).get_node_shared_ptr();
+
+        auto param_in_tensor = matched_node_shapeof->input(0).get_tensor_ptr();
+
+        // Create a new static Const instead of Param->ShapeOf
+        auto new_const = std::make_shared<ov::op::v0::Constant>(ov::element::i64,
+                                                                ov::Shape{param_in_tensor->get_shape().size()},
+                                                                param_in_tensor->get_shape().data());
```

Comment:
why `i64`? Also, what `get_shape().data()` returns to you?

Reply:
i64 is because op return this type. Should I get it instead of hard-code? `.data()` returns `void*`. `ov::Shape` is `std::vector`

Reply:
I guess I could omit `.data()` and just use shape as a vector